docs-user/bunny.md

```diff
@@ -0,0 +1,272 @@
+
+In a simplified example, the `worker.js` would run something like:
+
+```js
+// Evaluate the user's code string:
+eval(`
+rect('#fff', 10, 10, 3, 3);
+rect('#fff', 20, 20, 3, 3);
+rect('#fff', 30, 30, 3, 3);
+rect('#fff', 40, 40, 3, 3);
+rect('#fff', 50, 50, 3, 3);
+`);
+
+// worker.js would post something like this to the iframe:
+self.postMessage({
+  color: ['#fff', '#fff', '#fff', '#fff', '#fff'],
+  x: [10, 20, 30, 40, 50],
+  y: [10, 20, 30, 40, 50],
+  w: [3, 3, 3, 3, 3],
+  h: [3, 3, 3, 3, 3],
+});
+```
+
+Then the iframe would draw the code using:
+
+```js
+worker.addEventListener('message', message => {
+  const { data } = message;
+  for (let i = 0; i < data.color.length; i++) {
+    ctx.fillStyle = data.color[i];
+    ctx.fillRect(data.x[i], data.y[i], data.w[i], data.h[i]);
+  }
+});
+```
+
+This would result in user's evaluated code drawing something to the screen, like the bunny model in the above image.
+
+## The problem
+
+This code ended up not scaling well for large sets of rectangles being drawn to the screen. There were lots of stutters, and a slow frame rate. The user impact for fixing this problem would be to have a smoother frame rate, plus the ability to draw many more rectangles to the screen without slowing things down. In order to validate fixes, the following steps were used to reproduce the issue.
+
+ * Load the page with the bunny visualization.
+ * Hit Ctrl Shift 1 to turn on the Gecko Profiler.
+ * Wait around 5 seconds
+ * Hit Ctrl Shift 2 to capture the profile.
+ * Set the range to 3.0 seconds of relatively stable frames that don't have stutters or GC pauses.
+ * Hide idle stacks by right clicking `__psync_cvwait` and `mach_msg_trap` in the Flame Graph, and choosing "Drop samples with this function".
+ * Filter the threads to:
+   * The relevant content process
+   * The relevant DOM Worker
+   * The compositor
+
+This resulted in the following profile for the baseline profile: [https://perfht.ml/2IxTwqi](https://perfht.ml/2IxTwqi)
+
+### Getting oriented
+
+A nice place to start out is getting oriented with the thread stack graphs. These are in the headers and show the height of the stacks for the sample code. Keep in mind that a higher stack doesn't mean the code took longer to execute. Time is the X axis in this graph. There are gaps between the stacks. These gaps are the idle stacks that were hidden in the steps to reproduce.
```

Comment:
> Keep in mind that a higher stack doesn't mean the code took longer to execute.

Can you recall here what it means ? Also maybe insert new paragraphs before "keep in mind", and before "time is the X axis".